### PR TITLE
feat: add 360dialog service for sending whatsapp notification

### PR DESCRIPTION
**Pull request by xixas**
This PR adds and implements the 360dialog service functionality for sending whatsapp messages.

Sample request body:
```json
{
    "channelType": 3,
    "data": {
        "to": "919004812051",
        "type": "template",
        "template": {
            "namespace": "d8bcb6bd_2ab2_439c_9d9e_947501266c77",
            "name": "ir_incident_resolution",
            "language": {
                "policy": "deterministic",
                "code": "en"
            },
            "components": [
                {
                    "type": "body",
                    "parameters": [
                        {
                            "type": "text",
                            "text": "Bishal Biswas"
                        },
                        {
                            "type": "text",
                            "text": "WNK227"
                        },
                        {
                            "type": "text",
                            "text": "Massive Earthquake"
                        },
                        {
                            "type": "text",
                            "text": "Power Grid"
                        },
                        {
                            "type": "text",
                            "text": "Bishal Mondal"
                        },
                        {
                            "type": "text",
                            "text": "Vikas"
                        },
                        {
                            "type": "text",
                            "text": "Open"
                        },
                        {
                            "type": "text",
                            "text": "10"
                        },
                        {
                            "type": "text",
                            "text": "15755"
                        }
                    ]
                }
            ]
        }
    }
}
```

Sample response:
```json
{
    "status": "success",
    "data": {
        "notification": {
            "channelType": 3,
            "data": {
                "to": "919004812051",
                "type": "template",
                "template": {
                    "namespace": "d8bcb6bd_2ab2_439c_9d9e_947501266c77",
                    "name": "ir_incident_resolution",
                    "language": {
                        "policy": "deterministic",
                        "code": "en"
                    },
                    "components": [
                        {
                            "type": "body",
                            "parameters": [
                                {
                                    "type": "text",
                                    "text": "Bishal Biswas"
                                },
                                {
                                    "type": "text",
                                    "text": "WNK227"
                                },
                                {
                                    "type": "text",
                                    "text": "Massive Earthquake"
                                },
                                {
                                    "type": "text",
                                    "text": "Power Grid"
                                },
                                {
                                    "type": "text",
                                    "text": "Bishal Mondal"
                                },
                                {
                                    "type": "text",
                                    "text": "Vikas"
                                },
                                {
                                    "type": "text",
                                    "text": "Open"
                                },
                                {
                                    "type": "text",
                                    "text": "10"
                                },
                                {
                                    "type": "text",
                                    "text": "15755"
                                }
                            ]
                        }
                    ]
                }
            },
            "createdBy": "admin",
            "updatedBy": "admin",
            "result": null,
            "id": 5,
            "deliveryStatus": 1,
            "createdOn": "2023-10-19T05:32:56.000Z",
            "updatedOn": "2023-10-19T05:32:56.000Z",
            "status": 1
        }
    }
}
```

**Pending Actions**
- Have made changes to remove validation on data in body.

**Screenshots**
image

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,13 +24,15 @@
...
   "dependencies": {
+    "@nestjs/axios": "^3.0.0",
     "@nestjs/bull": "^10.0.1",
     "@nestjs/common": "^10.0.0",
     "@nestjs/config": "^3.0.0",
     "@nestjs/core": "^10.0.0",
     "@nestjs/platform-express": "^10.0.0",
     "@nestjs/schedule": "^3.0.3",
     "@nestjs/typeorm": "^10.0.0",
+    "axios": "^1.5.1",
```
Comment (osm-vishnukyatannawar) [inline package.json:35]: How did we make API call for mailgun if we had to add this now?

Comment (osm-vishnukyatannawar) [inline package.json:35]: Also do we need this when we have `"@nestjs/axios"`?

**Review by osm-vishnukyatannawar — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (xixas) [reply package.json:35]: We have Mailgun client provided by mailgun.js we were using that.

Comment (xixas) [reply package.json:35]: Yes, both need to be installed.

**Review by osm-vishnukyatannawar — APPROVED** [review]

**Pull request merged**